packages-explorer/src/gui/result.js

```diff
@@ -0,0 +1,245 @@
+import React from "react";
+import get from "lodash/get";
+import {use} from "../state";
+import FormattedLicense, {isUnfree} from "../license";
+import Link from "../link";
+
+/**
+ * All known platforms the widget can show.
+ * Further functions will filter the ones to show.
+ */
+const PLATFORMS = [
+	"x86_64-linux",
+	"aarch64-linux",
+	"x86_64-darwin",
+	"i686-linux",
+];
+
+const Result = ({
+	result,
+	result: {name, attr, meta: {description} = {}},
+	even,
+	selected,
+	select_attr,
+}) =>
+	<tr
+		class={[
+			"result",
+			even ? "even" : "odd",
+			selected === attr ? "is-selected" : "is-not-selected",
+			isUnfree(result["meta"]["license"]) ? "is-unfree" : "is-free",
+		].join(" ")}
+		onClick={() => select_attr(attr)}
+	>
+		<td>
+			<button
+				onClick={(e) => {
+					e.stopPropagation();
+					select_attr(attr);
+				}}
+			>{name}</button>
+		</td>
+		<td>{attr}</td>
+		<td>{description}</td>
+	</tr>
+;
+
+export default use(["selected"], ["select_attr"], Result);
+
+const NotSpecified = () => <em>Not specified</em>;
+const hydraLink = (attribute, platform, branch) => `https://hydra.nixos.org/job/${branch}/${attribute}.${platform}`;
+const githubLink = (commit, position) => `https://github.com/NixOS/nixpkgs/blob/${commit}/${position.replace(":", "#L")}`;
+
+const Install = use(["channel"], [], ({result, channel}) =>
+	<tr>
+		<th>Install command</th>
+		<td>
+			<tt>
+				<span class="command">
+					nix-env -iA {channel.replace(/-.*/, "")}.<span class="attrname">{result["attr"]}</span>
+				</span>
+			</tt>
+			{" "}<em class="muted">(NixOS channel)</em>
```

Comment:
This comes  from when everything was always written in terms of the NixOS channel: `nix-env -iA nixos._1password` it isn't particularly desirable stylistically :) Since yours fixes The Bug, we can delete it.

```suggestion
```